### PR TITLE
Return `SymbolMatch` rather than `FileMatchResolver`

### PR DESCRIPTION
**Pull request by camdencheek**
This refactors `searchZoektSymbols` and `computeSymbols` to 
return `[]*result.SymbolMatch` instead of `[]symbolResolver`. This
helps make them independent of GraphQL resolvers so the search logic
can be split out of `graphqlbackend`. 

There are 3 commits, each small and independent:
- First refactors `searchZoektSymbols`
- Second refactors `computeSymbols`, which uses `searchZoektSymbols`
- Third modifies `FileMatchResolver` to use a helper function added for this PR

### CODE REVIEW

File: cmd/frontend/graphqlbackend/symbols.go [hunk 1]
```diff
@@ -28,19 +28,33 @@ type symbolsArgs struct {
 }
 
 func (r *GitTreeEntryResolver) Symbols(ctx context.Context, args *symbolsArgs) (*symbolConnectionResolver, error) {
-	symbols, err := computeSymbols(ctx, r.db, r.commit, args.Query, args.First, args.IncludePatterns)
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/symbols.go:31]: removing `r.db`, love it!

File: cmd/frontend/graphqlbackend/symbols.go [hunk 2]
```diff
@@ -28,19 +28,33 @@ type symbolsArgs struct {
...
+	symbols, err := computeSymbols(ctx, r.commit, args.Query, args.First, args.IncludePatterns)
 	if err != nil && len(symbols) == 0 {
 		return nil, err
 	}
-	return &symbolConnectionResolver{symbols: symbols, first: args.First}, nil
+	return &symbolConnectionResolver{
+		symbols: symbolResultsToResolvers(r.db, r.commit, symbols),
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/symbols.go:36]: non-blocking: can we make this take a `SymbolMatch` rather? Then when the resolver is actually called we convert it into symbol resolvers? That will make the construction sites cleaner and avoid duplication?

**Review by keegancsmith — APPROVED** [review]
Love it!

File: cmd/frontend/graphqlbackend/symbols.go (hunk 2 above)
Comment (camdencheek) [reply cmd/frontend/graphqlbackend/symbols.go:36]: The issue here is then, we'd need to store the `db` and `commit` on the `symbolConnectionResolver` in order to construct them. We can totally do that, but it adds its own complexity. 

On a related note, I'm pretty unsure what the `symbolConnection resolver` actually does other than limiting. I understand that connections are intended for paging, but I don't see anywhere that we generate a cursor or mutate the remaining set of symbols.

Comment (camdencheek) [reply cmd/frontend/graphqlbackend/symbols.go:36]: I'll merge this as is for now, but I'm happy to continue the conversation here.

**Pull request merged**